cmd/util/cmd/epochs/cmd/recover.go

```diff
@@ -0,0 +1,186 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/flow-go/cmd/bootstrap/run"
+	"github.com/onflow/flow-go/cmd/util/cmd/common"
+	epochcmdutil "github.com/onflow/flow-go/cmd/util/cmd/epochs/utils"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol/inmem"
+)
+
+// generateRecoverEpochTxArgsCmd represents a command to generate the data needed to submit an epoch recovery transaction the network is in EFM (epoch fallback mode).
+// EFM can be exited only by a special service event, EpochRecover, which initially originates from a manual service account transaction.
+// The full epoch data must be generated manually and submitted with this transaction in order for an
+// EpochRecover event to be emitted. This command retrieves the current protocol state identities, computes the cluster assignment using those
+// identities, generates the cluster QC's and retrieves the DKG key vector of the last successful epoch.
+var (
+	generateRecoverEpochTxArgsCmd = &cobra.Command{
+		Use:   "generate-efm-recovery-data",
+		Short: "Generates recover epoch transaction arguments",
+		Long:  "Generates transaction arguments for the epoch recovery transaction.",
+		Run:   generateRecoverEpochTxArgs,
+	}
+
+	flagAnAddress               string
+	flagAnPubkey                string
+	flagPartnerWeights          string
+	flagPartnerNodeInfoDir      string
+	flagInternalNodePrivInfoDir string
+	flagConfig                  string
+	flagCollectionClusters      int
+	flagStartView               uint64
+	flagStakingEndView          uint64
+	flagEndView                 uint64
+)
+
+func init() {
+	rootCmd.AddCommand(generateRecoverEpochTxArgsCmd)
+	addGenerateRecoverEpochTxArgsCmdFlags()
+}
+
+func addGenerateRecoverEpochTxArgsCmdFlags() {
+	generateRecoverEpochTxArgsCmd.Flags().StringVar(&flagBucketNetworkName, "bucket-network-name", "",
+		"when retrieving the root snapshot from a GCP bucket, the network name portion of the URL (eg. \"mainnet-13\")")
+	generateRecoverEpochTxArgsCmd.Flags().IntVar(&flagCollectionClusters, "collection-clusters", 0,
```

Comment:
```suggestion
	generateRecoverEpochTxArgsCmd.Flags().IntVar(&flagCollectionClusters, "collection-clusters", 3,
```
Default to the Mainnet value